pkg/imageengine/buildah/build.go

```diff
@@ -0,0 +1,372 @@
+		if excludes, _, err = parse.ContainerIgnoreFile(contextDir, wrapper.IgnoreFile); err != nil {
+			return define.BuildOptions{}, []string{}, err
+		}
+	}
+
+	var timestamp *time.Time
+	if engine.Command.Flag("timestamp").Changed {
+		t := time.Unix(wrapper.Timestamp, 0).UTC()
+		timestamp = &t
+	}
+
+	compression := define.Gzip
+	if wrapper.DisableCompression {
+		compression = define.Uncompressed
+	}
+
+	options := define.BuildOptions{
+		AddCapabilities: wrapper.CapAdd,
+		AdditionalTags:  tags,
+		AllPlatforms:    wrapper.AllPlatforms,
+		Annotations:     wrapper.Annotation,
+		Architecture:    systemContext.ArchitectureChoice,
+		//Args:                    args,
+		BlobDirectory:           wrapper.BlobCache,
+		CNIConfigDir:            wrapper.CNIConfigDir,
+		CNIPluginPath:           wrapper.CNIPlugInPath,
+		CommonBuildOpts:         commonOpts,
+		Compression:             compression,
+		ConfigureNetwork:        networkPolicy,
+		ContextDirectory:        contextDir,
+		DefaultMountsFilePath:   "",
+		Devices:                 wrapper.Devices,
+		DropCapabilities:        wrapper.CapDrop,
+		Err:                     stderr,
+		ForceRmIntermediateCtrs: wrapper.ForceRm,
+		From:                    wrapper.From,
+		IDMappingOptions:        idmappingOptions,
+		IIDFile:                 wrapper.Iidfile,
+		In:                      stdin,
+		Isolation:               isolation,
+		IgnoreFile:              wrapper.IgnoreFile,
+		Labels:                  wrapper.Label,
+		Layers:                  layers,
+		LogRusage:               wrapper.LogRusage,
+		Manifest:                wrapper.Manifest,
+		MaxPullPushRetries:      maxPullPushRetries,
+		NamespaceOptions:        namespaceOptions,
+		NoCache:                 wrapper.NoCache,
+		OS:                      systemContext.OSChoice,
+		Out:                     stdout,
+		Output:                  output,
+		OutputFormat:            format,
+		PullPolicy:              pullPolicy,
+		PullPushRetryDelay:      pullPushRetryDelay,
+		Quiet:                   wrapper.Quiet,
+		RemoveIntermediateCtrs:  wrapper.Rm,
+		ReportWriter:            reporter,
+		Runtime:                 wrapper.Runtime,
+		RuntimeArgs:             runtimeFlags,
+		RusageLogFile:           wrapper.RusageLogFile,
+		SignBy:                  wrapper.SignBy,
+		SignaturePolicyPath:     wrapper.SignaturePolicy,
+		Squash:                  wrapper.Squash,
+		SystemContext:           systemContext,
+		Target:                  wrapper.Target,
+		TransientMounts:         wrapper.Volumes,
+		Jobs:                    &wrapper.Jobs,
+		Excludes:                excludes,
+		Timestamp:               timestamp,
+		Platforms:               platforms,
+		UnsetEnvs:               wrapper.UnsetEnvs,
+	}
+
+	if wrapper.Quiet {
+		options.ReportWriter = ioutil.Discard
+	}
+
+	return options, kubefiles, nil
+}
+
+func (engine *Engine) build(cxt context.Context, kubefiles []string, options define.BuildOptions) (id string, err error) {
```

Comment:
I found that this function always returns a nil-error. Is it reasonable? @justadogistaken